server/src/test/java/org/opensearch/action/admin/indices/forcemerge/ForceMergeRequestTests.java

```diff
@@ -31,21 +31,137 @@
 
         request = new ForceMergeRequest();
         request.maxNumSegments(12);
         request.onlyExpungeDeletes(true);
         request.flush(false);
-        assertEquals("Force-merge indices [], maxSegments[12], onlyExpungeDeletes[true], flush[false]", request.getDescription());
+        request.primaryOnly(true);
+        assertEquals(
+            "Force-merge indices [], maxSegments[12], onlyExpungeDeletes[true], flush[false], primaryOnly[true]",
+            request.getDescription()
+        );
+    }
+
+    public void testToString() {
+        ForceMergeRequest request = new ForceMergeRequest();
+        assertEquals("ForceMergeRequest{maxNumSegments=-1, onlyExpungeDeletes=false, flush=true, primaryOnly=false}", request.toString());
+
+        request = new ForceMergeRequest();
+        request.maxNumSegments(12);
+        request.onlyExpungeDeletes(true);
+        request.flush(false);
+        request.primaryOnly(true);
+        assertEquals("ForceMergeRequest{maxNumSegments=12, onlyExpungeDeletes=true, flush=false, primaryOnly=true}", request.toString());
+    }
+
+    public void testSerialization() throws Exception {
+        final ForceMergeRequest request = randomRequest();
+        try (BytesStreamOutput out = new BytesStreamOutput()) {
+            request.writeTo(out);
+
+            final ForceMergeRequest deserializedRequest;
+            try (StreamInput in = out.bytes().streamInput()) {
+                deserializedRequest = new ForceMergeRequest(in);
+            }
+            assertEquals(request.maxNumSegments(), deserializedRequest.maxNumSegments());
+            assertEquals(request.onlyExpungeDeletes(), deserializedRequest.onlyExpungeDeletes());
+            assertEquals(request.flush(), deserializedRequest.flush());
+            assertEquals(request.primaryOnly(), deserializedRequest.primaryOnly());
+            assertEquals(request.forceMergeUUID(), deserializedRequest.forceMergeUUID());
+        }
+    }
+
+    public void testBwcSerialization() throws Exception {
+        {
+            final ForceMergeRequest sample = randomRequest();
+            final Version compatibleVersion = VersionUtils.randomCompatibleVersion(random(), Version.CURRENT);
+            try (BytesStreamOutput out = new BytesStreamOutput()) {
+                out.setVersion(compatibleVersion);
+                sample.writeTo(out);
+
+                final ForceMergeRequest deserializedRequest;
+                try (StreamInput in = out.bytes().streamInput()) {
+                    in.setVersion(Version.CURRENT);
+                    deserializedRequest = new ForceMergeRequest(in);
+                }
+
+                assertEquals(sample.maxNumSegments(), deserializedRequest.maxNumSegments());
+                assertEquals(sample.onlyExpungeDeletes(), deserializedRequest.onlyExpungeDeletes());
+                assertEquals(sample.flush(), deserializedRequest.flush());
+                if (compatibleVersion.onOrAfter(Version.V_2_12_0)) {
```

Comment:
@gaobinlong The changes looks good. Do you plan to change this to `2_13_0` and target for next release ?

Probably would be great to add a test in ForceMergeIT as well to trigger force merge with `primaryOnly` flag and assert on the shard count in the response.

Reply:
Actually my bad, the version needs to be set as `Version.V_3_0_0` to avoid breaking bwc test in backport PR and later you will need to create another PR (after backport to 2.x) to change the versions here and in rest api yml test file. You can see this PR as a reference: https://github.com/opensearch-project/OpenSearch/pull/12446/files

Reply:
Thanks, I've changed that, and I'll open another PR to update the supported version for this change after the backport PR is merged.